intake_stac/catalog.py

```diff
@@ -102,82 +102,91 @@ def serialize(self):
 
 class StacCatalog(AbstractStacCatalog):
     """
-    Intake Catalog represeting a STAC Catalog
-    https://github.com/radiantearth/stac-spec/blob/master/catalog-spec/catalog-spec.md
-
-    A Catalog that references a STAC catalog at some URL
-    and constructs an intake catalog from it, with opinionated
-    choices about the drivers that will be used to load the datasets.
-    In general, the drivers are:
-
-        - netcdf
-        - rasterio
-        - xarray_image
-        - textfiles
+    Maps Intake Catalog to a STAC Catalog
+    https://pystac.readthedocs.io/en/latest/api.html?#catalog-spec
     """
 
     name = 'stac_catalog'
-    _stac_cls = satstac.Catalog
+    _stac_cls = pystac.Catalog
 
     def _load(self):
         """
         Load the STAC Catalog.
         """
-        subcatalog = None
-        # load first sublevel catalog(s)
-        for subcatalog in self._stac_obj.children():
+        for subcatalog in self._stac_obj.get_children():
+            if isinstance(subcatalog, pystac.Catalog):
+                driver = StacCatalog
+            elif isinstance(subcatalog, pystac.Collection):
+                driver = StacCollection
+
             self._entries[subcatalog.id] = LocalCatalogEntry(
                 name=subcatalog.id,
                 description=subcatalog.description,
-                driver=StacCatalog,
+                driver=driver,  # recursive
                 catalog=self,
-                args={'stac_obj': subcatalog.filename},
+                args={'stac_obj': subcatalog.get_self_href()},
             )
 
-        if subcatalog is None:
-            # load items under last catalog
-            for item in self._stac_obj.items():
-                self._entries[item.id] = LocalCatalogEntry(
-                    name=item.id,
-                    description='',
-                    driver=StacItem,
-                    catalog=self,
-                    args={'stac_obj': item},
-                )
+        for item in self._stac_obj.get_items():
+            self._entries[item.id] = LocalCatalogEntry(
+                name=item.id,
+                description='',
+                driver=StacItem,
+                catalog=self,
+                args={'stac_obj': item},
+            )
 
     def _get_metadata(self, **kwargs):
         """
         Keep copy of all STAC JSON except for links
         """
-        metadata = self._stac_obj._data.copy()
+        # NOTE: why not links?
+        metadata = self._stac_obj.to_dict()
         del metadata['links']
         return metadata
 
 
+# NOTE: not sure we need a standalone collection?
+# what methods would be here that couldn't just be in StacCatalog?
```

Comment:
leaving in a separate StacCollection class seems fine, but we don't have any methods or attributes currently that differ from StacCatalog...

Reply:
The spec is actually changing where Collection doesn't inherit from Catalog. I'm not sure that is relevant to this discussion, as in pystac the inheritance will most likely persist. However Collections have a number of properties that Catalogs do not, most notably the spatial and temporal extents. If intake really doesn't need any collection properties it might make sense to treat them as catalogs, but IMO it would align better with the spec and potential future features to represent Collections as a separate entity.